client.go

```diff
@@ -1405,32 +1424,50 @@ func (c *Client[TTx]) InsertManyTx(ctx context.Context, tx TTx, params []InsertM
 	return c.insertFastMany(ctx, exec, insertParams)
 }
 
-func (c *Client[TTx]) insertFastMany(ctx context.Context, tx riverdriver.ExecutorTx, insertParams []*riverdriver.JobInsertFastParams) (int, error) {
-	inserted, err := tx.JobInsertFastMany(ctx, insertParams)
-	if err != nil {
-		return inserted, err
-	}
+func (c *Client[TTx]) insertFastMany(ctx context.Context, tx riverdriver.ExecutorTx, manyParams []*rivertype.JobInsertParams) (int, error) {
+	doInner := func(ctx context.Context) (int, error) {
+		manyInsertParams := sliceutil.Map(manyParams, func(params *rivertype.JobInsertParams) *riverdriver.JobInsertFastParams {
+			return (*riverdriver.JobInsertFastParams)(params)
```

Comment:
I worry a bit about the number of places where we have to do this kind of struct pointer casting and the potential danger of it. A lot of those are in tests and presumably any issues would be caught by tests, but I still felt like I should point out that it makes me feel a little uneasy.

Reply:
I guess the compiler actually catches these, so they're nbd?

Reply:
Yeah exactly. The structs have to match _exactly_ or the compiler won't let you do it. Even having a non-comparable field in there like a `[]byte` will prevent this.

FWIW, I don't worry two much about two separate structs that are set manually from each other, but I figure as long as they're exactly identical, may as well cast from one to the other.